src/ranch_ssl.erl

```diff
@@ -296,3 +297,18 @@ cleanup(#{socket_opts:=SocketOpts}) ->
 	end;
 cleanup(_) ->
 	ok.
+
+-spec strip_usupported_options(opts()) -> opts().
+strip_usupported_options(SocketOpts) ->
+    case lists:keyfind(versions, 1, SocketOpts) of
+        {versions, ['tlsv1.3']} ->
+            Intermediate1 = lists:keydelete(secure_renegotiate, 1, SocketOpts),
+            Intermediate2 = lists:keydelete(reuse_sessions, 1, Intermediate1),
+            Intermediate3 = lists:keydelete(next_protocols_advertised, 1, Intermediate2),
+            lists:keydelete(alpn_preferred_protocols, 1, Intermediate3);
+        _ ->
+            SocketOpts
+    end;
+strip_usupported_options(SocketOpts) ->
```

Comment:
This clause is pointless, it will never be reached.

Reply:
I'll remove this. It was inherited from the code, which I'd moved from the acceptors file.